pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -58,6 +59,9 @@ func (k *Kaniko) BuildAndPushContainerImage(buildOptions *BuildOptions, namespac
 		return errors.Wrap(err, "Failed to publish kaniko job")
 	}
 
+	// Cleanup
+	defer k.deleteJob(namespace, kanikoJob.Name) // nolint: errcheck
+
 	k.logger.Debug("Waiting for kaniko to finish")
 	timeout := time.Now().Add(10 * time.Minute)
 	for time.Now().Before(timeout) {
```

Comment:
I would move the below into a gocoroutine, and use a boolean channel to determine whether the kaniko job was done. if a timeout occur (`<-time.After(...)`) then follow a failure flow

That would be a bit more readable code imo

Something like described [here](https://gobyexample.com/timeouts)

Reply:
I'm not sure why that would be more readable. If you're creating a goroutine and waiting on it... why create a go routine? Go encourages synchronous code because it is assumed it runs in its own execution context. 

I would, however, move the contents of the for loop to a helper function